2-repositories/iglu-server/src/main/scala/com/snowplowanalytics/iglu/server/Server.scala

```diff
@@ -101,7 +104,14 @@ object Server {
       maxAge = 1.day.toSeconds
     )
     val serverRoutes = (if (debug) debugRoute :: routes else routes).map {
-      case (endpoint, route) => (endpoint, CORS(AutoSlash(route), corsConfig))
+      case (endpoint, route) =>
+        if (debug)
+          (endpoint, Logger.httpRoutes[IO](logBody = true, logHeaders = true, redactHeadersWhen = (Headers.SensitiveHeaders + "apikey".ci).contains, logAction = Some(s => for {
```

Comment:
Couple of points here:

1. I think you're greatly overusing lambda syntax, e.g.

```scala
Some(s => for {
  logger <- Slf4jLogger.create[IO]
  _ <- logger.info(s)
} yield ()))
```

It will be **much** easier to read if it will be two separate constructs:

```scala
def createLogger(s: Whatever): F[Unit] =
  for {
    logger <- Slf4jLogger.create[IO]
    _ <- logger.info(s)
  } yield ()

Some(createLogger)
```

Try to avoid putting everything in paretheses.

2. Logging libraries (to which log4cats provides an interface) work with their own system properties configuration (like this https://stackoverflow.com/questions/14544991/how-to-configure-slf4j-simple). You decide what level the message belongs to, devops decides whether they want to see messages of this detalization. As an example you can use BigQuery Repeater https://github.com/snowplow-incubator/snowplow-bigquery-loader/pull/22/commits/291bcbf65b26cb464d4f8e60fc5a8e0dc478ad6e#diff-7b03362c42f06ee6f096031100f0db76R45